variables-addon-irsa.tf

```diff
@@ -0,0 +1,93 @@
+# IMPORTANT: This file is synced with the "terraform-aws-eks-universal-addon" module. Any changes to this file might be overwritten upon the next release of that module.
+
+# ================ IRSA variables (optional) ================
+
+variable "cluster_identity_oidc_issuer" {
+  type        = string
+  default     = null
+  description = "The OIDC Identity issuer for the cluster (required)."
+}
+
+variable "cluster_identity_oidc_issuer_arn" {
+  type        = string
+  default     = null
+  description = "The OIDC Identity issuer ARN for the cluster that can be used to associate IAM roles with a Service Account (required)."
+}
+
+variable "rbac_create" {
+  type        = bool
+  default     = null
+  description = "Whether to create and use RBAC resources. Defaults to `true`."
+}
+
+variable "service_account_create" {
+  type        = bool
+  default     = null
+  description = "Whether to create Service Account. Defaults to `true`."
+}
+
+variable "service_account_name" {
+  type        = string
+  default     = null
+  description = "The Kubernetes Service Account name. Defaults to addon name."
+}
+
+variable "service_account_namespace" {
+  type        = string
+  default     = null
+  description = "The Kubernetes Service Account namespace. Defaults to addon namespace."
+}
+
+variable "irsa_role_create" {
+  type        = bool
+  default     = null
+  description = "Whether to create IRSA role and annotate Service Account. Defaults to `true`."
+}
+
+variable "irsa_role_name_prefix" {
+  type        = string
+  default     = null
+  description = "IRSA role name prefix. Defaults to addon IRSA component name with `irsa` suffix."
+}
+
+variable "irsa_role_name" {
+  type        = string
+  default     = null
+  description = "IRSA role name. The value is prefixed by `var.irsa_role_name_prefix`. Defaults to addon Helm chart name."
+}
+
+variable "irsa_policy_enabled" {
+  type        = bool
+  default     = null
+  description = "Whether to create IAM policy specified by `irsa_policy`. Mutually exclusive with `irsa_assume_role_enabled`. Defaults to `false`."
+}
+
+variable "irsa_policy" {
+  type        = string
+  default     = null
+  description = "Policy to be attached to the default role. Applied only if `irsa_policy_enabled` is `true`. Defaults to `\"\"`."
+}
```

Comment:
How would this look, for example, with [this repository](https://github.com/lablabs/terraform-aws-eks-load-balancer-controller/blob/main/iam.tf#L5) which has an opinionated policy? Because then the premise stated in the `irsa_policy` description does not apply at all.

Reply:
Removed `irsa_policy_enabled` and `irsa_policy` whatsoever. If some addon provides its opinionated policy it will pass it to the addon-irsa module via `irsa_additional_policies` variable by merging `var.irsa_additional_policies` with a locally created policy and passing it down to the IRSA module. Example:

```
addon_irsa = {
  (local.addon.name) = {
     irsa_additional_policies = merge({ (local.addon.name) = aws_iam_policy.opinionated_policy.arn }, var.irsa_additional_policies)
  }
}
```

Reply:
I've reverted `irsa_policy` behaviour and polished the description little bit.